### DISCUSSION TITLE
RTMP or RTSP

### DISCUSSION THREAD

**Discussion by TarcisioMenezes** [fwestenberg/reolink_dev#471] [Q&A] [answered]
Hey guys!
What should I take in consideration when choosing between RTMP or RTSP? What are their pros and cons? 
Don't know which one to choose. 
Thank you!

**Comment by tommy5c** [comment, marked as answer]
I think it is a matter of how you plan to use it. RTSP is typically a {you log into source} type of video meaning more of a pull style request
whereas RTMP is a {i am sending the source to...} and is a push request. that's not exactly the way it works but close enough. basically would you like to have a system that you log into to see or would you like to send the feed elsewhere/ I hope that helps

**Comment by xannor** [comment]
I would like to add to this due to my recent conversation with REOLink tech support.

Their camera's RTMP implementation does not support H265 so you can only use RTSP to view the main feed of any 4K+ camera. Even the web interface will downgrade to the sub stream when you try to view a recording. So a second answer to your question is, if you have 4K cameras and want to view the full pixel feed, you must use RTSP, otherwise RTMP is fine. RTMP also tends to be a little slower with the initial startup, so if you wanted jump to a live feed it has a few second delay, RTSP seems to be a little faster in this regard.
